sklearn/cluster/mean_shift_.py

```diff
@@ -143,28 +176,15 @@ def mean_shift(X, bandwidth=None, seeds=None, bin_seeding=False,
         else:
             seeds = X
     n_samples, n_features = X.shape
-    stop_thresh = 1e-3 * bandwidth  # when mean has converged
     center_intensity_dict = {}
     nbrs = NearestNeighbors(radius=bandwidth).fit(X)
 
-    # For each seed, climb gradient until convergence or max_iter
-    for my_mean in seeds:
-        completed_iterations = 0
-        while True:
-            # Find mean of points within bandwidth
-            i_nbrs = nbrs.radius_neighbors([my_mean], bandwidth,
-                                           return_distance=False)[0]
-            points_within = X[i_nbrs]
-            if len(points_within) == 0:
-                break  # Depending on seeding strategy this condition may occur
-            my_old_mean = my_mean  # save the old mean
-            my_mean = np.mean(points_within, axis=0)
-            # If converged or at max_iter, adds the cluster
-            if (extmath.norm(my_mean - my_old_mean) < stop_thresh or
-                    completed_iterations == max_iter):
-                center_intensity_dict[tuple(my_mean)] = len(points_within)
-                break
-            completed_iterations += 1
+    #execute iterations on all seeds in parallel
+    all_res = Parallel(n_jobs=n_jobs)(delayed(_mean_shift_single_seed)(seed,X,nbrs,max_iter) for seed in seeds)
```

Comment:
style: you can break the line after `Parallel(n_jobs=n_jobs)(` and before `delayed(...)`